src/reducers/dialogVisibility.js

```diff
@@ -0,0 +1,25 @@
+import Immutable from 'immutable'
+import { actionReducer } from './utils'
+
+const dialogVisibility = actionReducer(Immutable.fromJS({ showVmDialog: false, showVmDetail: false, showEditTemplate: false }), {
```

Comment:
yes, this representation is a call for trouble because it is a question of time until due to some bug you will have {showVmDialog: true, showVmDetail: true, showEditTemplate: true} - and now what :)

The struct: {type: ..., behavior: ...} sounds good to me. Could be: {type: VM, behavior: edit}